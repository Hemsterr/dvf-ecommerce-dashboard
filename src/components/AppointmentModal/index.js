// @flow
// Libs
import React from 'react'

// Components
import Modal from '../Modal'

type Props = {
  handleCloseModal: Function,
}

const bookingLink = 'https://calendly.com/vanessapatel/virtual-fitting'

class AppointmentModal extends React.PureComponent<Props> {
  static defaultProps = {
    handleCloseModal: () => {},
  }

  render() {
    const { handleCloseModal } = this.props

    return (
      <Modal className="shipping-modal" handleCloseModal={handleCloseModal}>
        <div className="shipping-modal">
          <p className="shipping-modal__title">
            Ready to book your appointment?
          </p>
          <div className="shipping-modal__appointmentWrapper">
            <a
              target="_blank"
              rel="noopener noreferrer"
              href={bookingLink}
              className="btn btn__success btn__small shipping-modal__appointment"
            >
              Book an appointment
            </a>
          </div>
        </div>
      </Modal>
    )
  }
}

export default AppointmentModal
