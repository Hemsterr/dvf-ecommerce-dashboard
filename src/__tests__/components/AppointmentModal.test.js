// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Component
import AppointmentModal from '../../components/AppointmentModal'

const props = {
  handleCloseModal: jest.fn(),
}

const wrapper = (props = {}) => {
  const component = shallow(<AppointmentModal {...props} />)

  return component
}

describe('AppointmentModal component', () => {
  const wrapperWithProps = wrapper(props)

  // Snapshot
  test('should render correctly AppointmentModal component default props', () => {
    expect(shallowToJson(wrapperWithProps)).toMatchSnapshot()
    expect(AppointmentModal.defaultProps.handleCloseModal()).toBeUndefined()
  })
})
