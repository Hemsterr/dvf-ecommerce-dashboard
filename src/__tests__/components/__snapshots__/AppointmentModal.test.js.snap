// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentModal component should render correctly AppointmentModal component default props 1`] = `
<Modal
  className="shipping-modal"
  handleCloseModal={[MockFunction]}
  title=""
>
  <div
    className="shipping-modal"
  >
    <p
      className="shipping-modal__title"
    >
      Ready to book your appointment?
    </p>
    <div
      className="shipping-modal__appointmentWrapper"
    >
      <a
        className="btn btn__success btn__small shipping-modal__appointment"
        href="https://booking.mangomint.com/852173"
        target="blank"
      >
        Book an appointment
      </a>
    </div>
  </div>
</Modal>
`;
